tftypes/refinement/string_prefix.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package refinement
+
+import "fmt"
+
+// StringPrefix represents an unknown value refinement that indicates the final value will be prefixed with the specified string value.
+// String prefixes that exceed 256 characters in length will be truncated and empty string prefixes will not be encoded. This refinement can
+// only be applied to the String type.
+type StringPrefix struct {
+	value string
+}
+
+func (s StringPrefix) Equal(other Refinement) bool {
+	otherVal, ok := other.(StringPrefix)
+	if !ok {
+		return false
+	}
+
+	return s.PrefixValue() == otherVal.PrefixValue()
+}
+
+func (s StringPrefix) String() string {
+	return fmt.Sprintf("prefix = %q", s.PrefixValue())
+}
+
+// PrefixValue returns the string value that the final value will be prefixed with.
+func (s StringPrefix) PrefixValue() string {
+	return s.value
+}
+
+func (s StringPrefix) unimplementable() {}
+
+// NewStringPrefix returns the StringPrefix unknown value refinement that indicates the final value will be prefixed with the specified
+// string value. String prefixes that exceed 256 characters in length will be truncated and empty string prefixes will not be encoded. This
+// refinement can only be applied to the String type.
+func NewStringPrefix(value string) Refinement {
+	return StringPrefix{
+		value: value,
+	}
+}
```

Comment:
Thanks for sharing that @apparentlymart! I've put this work down ATM, but I'll make a note for later to come back and decide what we want to do here. Like you said, at the very least documentation should be added.

I did run a quick test and Terraform does check the refinements during it's plan/apply data consistency check, so that's good!

![image](https://github.com/user-attachments/assets/f671b80c-2026-4677-a396-076733b7b89e)
![image](https://github.com/user-attachments/assets/0cdc463e-58f4-44a5-8428-baa75767dd5f)

```bash
--- FAIL: TestSchemaResource_basic (0.89s)
    /Users/austin.valle/code/terraform-provider-corner/internal/framework6provider/schema_resource_test.go:19: Step 1/1 error: Error running apply: exit status 1
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to framework_schema.test, provider
        "provider[\"registry.terraform.io/hashicorp/framework\"]" produced an
        unexpected new value: .prefix_test: final value cty.StringVal("hí") does not
        conform to planning placeholder
        cty.UnknownVal(cty.String).Refine().NotNull().StringPrefixFull("hi").NewValue().
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
```